test/FunctionalTests/Tests/DataOData/Tests/OData.Writer.Tests/JsonLight/JsonLightEntryWriterTests.cs

```diff
@@ -432,14 +432,104 @@ public void ActionAndFunctionPayloadOrderTest()
+                    {
+                        TypeName = "TestModel.CustomerType",
+                        Properties = new ODataProperty[]
+                        {
+                            new ODataProperty { Name = "ID", Value = (int)42 },
+                            new ODataProperty { Name = "Hobby", Value = "Hiking" },
+                        }
+                    } },
+                    Model = model,
+                    EntitySet = customerSet,
+                    Json = string.Join("$(NL)",
+                        "{{",
+                        "{0}\"ID\":\"42\", \"Hobby\":\"Hiking\"",
+                        "}}")
+                },
+                new EntryPayloadTestCase
+                {
+                    DebugDescription = "Customer instance without Hobby.",
+                    Items = new[] {new ODataResource()
+                    {
+                        TypeName = "TestModel.CustomerType",
+                        Properties = new ODataProperty[]
+                        {
+                            new ODataProperty { Name = "ID", Value = (int)44 },
+                            new ODataProperty { Name = "Hobby", Value = null },
+                        }
+                    } },
+                    Model = model,
+                    EntitySet = customerSet,
+                    Json = string.Join("$(NL)",
+                        "{{",
+                        "{0}\"ID\":\"44\"",
+                        "}}")
+                },
+
+            };
+
+            IEnumerable<PayloadWriterTestDescriptor<ODataItem>> testDescriptors = testCases.Select(testCase =>
+                new PayloadWriterTestDescriptor<ODataItem>(
+                    this.Settings,
+                    testCase.Items,
+                    tc => new JsonWriterTestExpectedResults(this.Settings.ExpectedResultSettings)
+                    {
+                        Json = string.Format(
+                            CultureInfo.InvariantCulture,
+                            testCase.Json,
+                            JsonLightWriterUtils.GetMetadataUrlPropertyForEntry(testCase.EntitySet.Name) + ","),
+                        FragmentExtractor = (result) => result.RemoveAllAnnotations(true)
+                    })
+                {
+                    DebugDescription = testCase.DebugDescription,
+                    Model = testCase.Model,
+                    PayloadEdmElementContainer = testCase.EntitySet,
+                    PayloadEdmElementType = testCase.EntityType,
+                    SkipTestConfiguration = testCase.SkipTestConfiguration
+                });
+
+            this.CombinatorialEngineProvider.RunCombinations(
+                testDescriptors,
+                this.WriterTestConfigurationProvider.JsonLightFormatConfigurationsWithIndent,
+                (testDescriptor, testConfiguration) =>
+                {
+                    testConfiguration.MessageWriterSettings.OmitNullValues = true;

```

Comment:
Could you add EndToEnd test that uses  responseMessageWithModel.PreferenceAppliedHeader().OmitValues= "nulls"; to check e2e? Something like tests for AnnotationFilter in WritePayloadTests or InstanceAnnotationWriterIntegrationTests

Reply:
Will do.

Reply:
Added e2e test WriteJsonWittoutModelTests::ExpandedCustomerEntryOmitNullValuesTest with proper header and null values for various cases.